### PR TITLE
[LIGHT] No affordance ablation

### PR DESCRIPTION
**Pull request by JackUrb**
**Patch description**
Based on reviews, we're adding an ablation to this task for not having affordance information. This PR adds an option to drop affordance information, which ends up creating a list of all possible object interactions if all of the objects were to have all of the affordances. 

**Testing steps**
```
python examples/display_data.py -t light_dialog -dt test --light_label_type action --light_use_affordances none
```

**Logs**
```
_task_action
_setting_name Castle Maids' Room, Inside Castle
_setting_desc Inside the maids room is dark and grey.  It is much different from the rest of the Castle which has grand walls and floors and paintings from famous painters around the wall.  The maid holds many secrets in this room.  One which is a secret door that holds paintings she should not have.
_partner_name worker
_self_name painter
_self_persona I was hired to paint a portrait of the noblewoman the king would eventually marry. It seems that the painting was too flattering and when the king lifted the veil, he was very angry. I have been imprisoned for four long years. I had a wife and six children, but they do not wait for me to return. I turn my agony into art.
_object_desc a floor : THe floor is dusty and broken in ...
I am just a lowly worker. I think the red paint looks good. I can assist you with painting.
_partner_act get brush
_self_say I will love that. Tell me, how long have you been here?

[eval_labels: give brush to worker]
[label_candidates: remove fresh red paint|eat shirt|wield fresh red paint|drop shirt|drop fresh red paint|...and 37 more]
```

### CODE REVIEW

**Comment by JackUrb** [comment]
I believe datatests are broken:
```
python tests/datatests/test_new_tasks.py
Traceback (most recent call last):
  File "tests/datatests/test_new_tasks.py", line 60, in test_verify_data
    text, log = verify(opt, print_parser=False)
  File "/Users/jju/ParlAI/parlai/scripts/verify_data.py", line 120, in verify
    text, log = report(world, counts, log_time)
  File "/Users/jju/ParlAI/parlai/scripts/verify_data.py", line 44, in report
    'did_not_return_message': counts['returns_messages'],
KeyError: 'returns_messages'
Got above exception in light_dialog:DefaultTeacher
E
======================================================================
ERROR: test_verify_data (__main__.TestNewTasks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/datatests/test_new_tasks.py", line 66, in test_verify_data
    if log[key] != 0:
UnboundLocalError: local variable 'log' referenced before assignment

----------------------------------------------------------------------
Ran 1 test in 40.801s
```

Adding a fix

**Comment by JackUrb** [comment]
```
python tests/datatests/test_new_tasks.py
.
----------------------------------------------------------------------
Ran 1 test in 30.907s

OK
```

File: parlai/scripts/verify_data.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def report(world, counts, log_time):
         'label_candidates_with_missing_label': counts[
             'label_candidates_with_missing_label'
         ],
-        'did_not_return_message': counts['returns_messages'],
+        'did_not_return_message': counts['did_not_return_message'],
```
Comment (emilydinan) [inline parlai/scripts/verify_data.py]: thanks for the fix here!

File: parlai/tasks/light_dialog/agents.py [hunk 2]
```diff
@@ -86,6 +87,9 @@ def add_cmdline_args(argparser):
             default='all',
             choices=['partner', 'self', 'all', 'none'],
         )
+        agent.add_argument(
+            '--light_use_affordances', type=str, default='all', choices=['all', 'none']
```
Comment (emilydinan) [inline parlai/tasks/light_dialog/agents.py]: do you foresee there being more than two options in the future? if not, a more intuitive arg might be a bool `--light-use-all-affordances`.

**Pull request merged**